onnxruntime/core/optimizer/layer_norm_fusion.cc

```diff
@@ -212,36 +235,78 @@ Status LayerNormFusion::ApplyImpl(Graph& graph, bool& modified, int graph_level,
     }
     nodes_to_remove.push_back(last_add_node);
 
+    // verify axes is contiguous if more than one elements
+    const onnxruntime::NodeAttributes& attributes = reduce_mean_node.GetAttributes();
+    std::vector<int64_t> axes_values;
+    if (attributes.find("axes") != attributes.end()) {
+      axes_values = RetrieveValues<int64_t>(attributes.at("axes"));
+      if (axes_values.size() > 1) {
+        sort(axes_values.begin(), axes_values.end());
+        bool contingous = true;
+        for (size_t i = 0; i < axes_values.size() - 1; i++) {
+          if (axes_values[i] + 1 != axes_values[i + 1]) {
```

Comment:
This might not work in some cases when axis is negative. For example, data dim [a, b, c] with axes [1, -1].

Reply:
checked with Sherlock, when there are multiple axes, the ints should be contiguous for layernorm/reducemean.
I dump additional cases here to onnx model: 
[https://pytorch.org/docs/master/generated/torch.nn.LayerNorm.html#torch.nn.LayerNorm](https://pytorch.org/docs/master/generated/torch.nn.LayerNorm.html#torch.nn.LayerNorm)
all the axes are contiguous. Therefore I am adding the check. 


---
In reply to: [445769645](https://github.com/microsoft/onnxruntime/pull/4320#discussion_r445769645) [](ancestors = 445769645)

Reply:
I see. I'm asking because the range of "axes" could be [-r, r-1], r being the rank of data. Do you mean that for "axes" attribute, all values in the list should be either all positive or all negative? In theory reducing with axes [1, -1] on data rank of 3 like my example mentioned above is equivalent to [1, 2] and is also contiguous, but here will return false.

Reply:
I see your point, let me include this in the check logic. thx.